allensdk/brain_observatory/ecephys/static_gratings.py

```diff
@@ -0,0 +1,378 @@
+    def peak_columns(self):
+        return [c[0] for c in self.PEAK_COLS]
+
+    @property
+    def peak_dtypes(self):
+        return [c[1] for c in self.PEAK_COLS]
+
+    @property
+    def peak(self):
+        if self._peak is None:
+            # pandas can have issues interpreting type and makes the column 'object' type, this should enforce the
+            # correct data type for each column
+            peak_df = pd.DataFrame(np.empty(self.numbercells, dtype=np.dtype(self.PEAK_COLS)),
+                                   index=range(self.numbercells))
+
+            # set values to null by default
+            peak_df['fit_sf_sg'] = np.nan
+            peak_df['fit_sf_ind_sg'] = np.nan
+            peak_df['sf_low_cutoff_sg'] = np.nan
+            peak_df['sf_high_cutoff_sg'] = np.nan
+
+            # TODO: Make cell_id the df index?
+            peak_df['cell_specimen_id'] = list(self.spikes.keys())
+            peak_df['lifetime_sparseness_sg'] = self._get_lifetime_sparseness()
+
+            for nc, unit_id in enumerate(self.spikes.keys()):
+                peaks = np.where(self.response_events[:, 1:, :, nc, 0] == self.response_events[:, 1:, :, nc, 0].max())
+                pref_ori = peaks[0][0]
+                pref_sf = peaks[1][0]
+                pref_phase = peaks[2][0]
+
+                peak_df.loc[nc, 'pref_ori_sg'] = self.orivals[pref_ori]
+                peak_df.loc[nc, 'pref_sf_sg'] = self.sfvals[pref_sf]
+                peak_df.loc[nc, 'pref_phase_sg'] = self.phasevals[pref_phase]
+
+                peak_df.loc[nc, 'num_pref_trials_sg'] = int(self.response_events[pref_ori, pref_sf+1, pref_phase, nc, 2])
+                peak_df.loc[nc, 'responsive_sg'] = self.response_events[pref_ori, pref_sf+1, pref_phase, nc, 2] > 11
+
+                stim_table_mask = (self.stim_table['SF'] == self.sfvals[pref_sf]) & \
+                                  (self.stim_table['Ori'] == self.orivals[pref_ori]) & \
+                                  (self.stim_table['Phase'] == self.phasevals[pref_phase])
+                peak_df.loc[nc, 'g_osi_sg'] = self._get_osi(pref_sf, pref_phase, nc)
+
+                peak_df.loc[nc, 'reliability_sg'] = self._get_reliability(unit_id, stim_table_mask)
+                peak_df.loc[nc, 'sfdi_sg'] = self._get_sfdi(pref_ori, pref_phase, nc)
+                peak_df.loc[nc, ['run_pval_sg', 'run_mod_sg', 'run_resp_sg', 'stat_resp_sg']] = \
+                    self._get_running_modulation(unit_id, stim_table_mask)
+
+                if self.response_events[pref_ori, pref_sf+1, pref_phase, nc, 2] > 11:
+                    peak_df.loc[nc, ['fit_sf_ind_sg', 'fit_sf_sg', 'sf_low_cutoff_sg', 'sf_high_cutoff_sg']] = \
+                        self._fit_sf_tuning(pref_ori, pref_sf, pref_phase, nc)
+
+            self._peak = peak_df
+
+        return self._peak
+
+    def _get_stim_table_stats(self):
+        sg_stim_table = self.stim_table
+        self._orivals = np.sort(sg_stim_table['Ori'].dropna().unique())
```

Comment:
can we make Ori & other keys parameters of the constructor? We might want to change these.